crates/core/src/nats/streams.rs

```diff
@@ -0,0 +1,189 @@
+use std::collections::HashMap;
+
+use futures_util::future::try_join_all;
+use strum::IntoEnumIterator;
+use strum_macros::EnumIter;
+
+use super::{
+    client::NatsClient,
+    types::{
+        JetStreamConfig,
+        NatsConsumer,
+        NatsStorageType,
+        NatsStream,
+        PullConsumerConfig,
+    },
+    NatsError,
+    SubjectName,
+};
+
+#[derive(Debug, EnumIter, Clone, Hash, Eq, PartialEq)]
+pub enum StreamKind {
+    Blocks,
+    Transactions,
+}
+
+impl StreamKind {
+    pub fn get_subjects(&self, prefix: &str) -> Vec<String> {
+        match self {
+            Self::Blocks => vec![SubjectName::Blocks.with_prefix(prefix)],
+            Self::Transactions => vec![
+                SubjectName::Transactions.with_prefix(prefix),
+                SubjectName::TransactionsById.with_prefix(prefix),
+            ],
+        }
+    }
```

Comment:
Why do we need this? It's a leak with our Enum abstraction. If this is how we have to manage the subjects, then we should probably revisit the entire Subjects abstraction.